demo/test_demos.py

```diff
@@ -38,16 +42,33 @@ def test_demo(file, scalar_type):
         # Skip demos that are only implemented for complex scalars
         pytest.skip()
 
-    opts = f"--scalar_type {scalar_type}"
-    extra_flags = "-Wunused-variable -Werror -fPIC "
-    assert os.system(f"cd {demo_dir} && ffcx {opts} {file}.py") == 0
-    assert (
-        os.system(
-            f"cd {demo_dir} && "
-            "CPATH=../ffcx/codegeneration/ "
-            f"gcc -I/usr/include/python{sys.version_info.major}.{sys.version_info.minor} "
-            f"{extra_flags} "
-            f"-shared {file}.c -o {file}.so"
+    if sys.platform.startswith("win32"):
+        opts = f"--scalar_type {scalar_type}"
+        extra_flags = "/std:c17"
+        assert os.system(f"cd {demo_dir} && ffcx {opts} {file}.py") == 0
+        assert (
+            os.system(
+                f"cd {demo_dir} && " f'cl.exe /I "../ffcx/codegeneration" {extra_flags} /c {file}.c'
+            )
+        ) == 0
+        assert (
+            os.system(
+                f"cd {demo_dir} && "
+                f'clang-cl.exe /I "../ffcx/codegeneration" {extra_flags} /c {file}.c'
+            )
+        ) == 0
+    else:
+        opts = f"--scalar_type {scalar_type}"
+        extra_flags = (
+            "-std=c17 -Wunused-variable -Werror -fPIC -Wno-error=implicit-function-declaration"
+        )
+        assert os.system(f"cd {demo_dir} && ffcx {opts} {file}.py") == 0
+        assert (
+            os.system(
+                f"cd {demo_dir} && "
+                f"gcc -I../ffcx/codegeneration "
```

Comment:
```suggestion
                f"cc -I../ffcx/codegeneration "
```

Reply:
fwiw, I think `cc = os.environ.get("CC", sysconfig.get_config_var("CC"))` is more likely to get the right thing and accepts the ubiquitous standard env override.